pkg/controller/postgresql.go

```diff
@@ -420,6 +420,14 @@ func (c *Controller) queueClusterEvent(informerOldSpec, informerNewSpec *acidv1.
 		clusterError = informerNewSpec.Error
 	}
 
+	// only allow deletion if delete annotations are set and conditions are met
+	if eventType == EventDelete {
+		if err := c.meetsClusterDeleteAnnotations(informerOldSpec); err != nil {
+			c.logger.WithField("cluster-name", clusterName).Warnf("skipping %q event as delete criteria not fulfilled: %s", eventType, err)
```

Comment:
`skipping %q event as delete criteria not fulfilled: %s`
To
`Ignoring delete event for cluster %q - manifest does not fulfill delete requirements (annotations): %s`